streamy.py

```diff
@@ -15,7 +15,9 @@
 _MATCH_EXTRA_DATA_ERROR = re.compile(r'Extra data: .*\(char (\d+).*\)').match
 
 
-def get_end_pos(e):
+def _get_end_pos(e):
+    if getattr(e, 'msg') and not e.msg.startswith('Extra data'):
+        return
```

Comment:
Perhaps put an empty line after this return to be consistent with the rest of the code?